www/js/equation.js

```diff
@@ -313,6 +313,17 @@ define([
 			}
 			return true;
 		},
+		isOnlySum: function(parse){
+			// Return true if expression is a sum of variables, restricting minus signs.
+			// Used to display + sign on node if expression is merely a sum
+			var ops = parse.operators();
+			var allowed = {"+": true, "variable": true};
+			for(var op in ops){
+				if(ops[op] > 0 && !allowed[op])
+					return false;
```

Comment:
I think dharani this is happening because isSum is to be handled for ops[op] == 0 case. Because we only parse part of the equation and not everything. Hence there is an implicit "+" which is causing the issue. So you should be updating isSum function only.